### PR TITLE
Fix started_at/finished_at always being nil

### PR DESCRIPTION
**Pull request by agrare**
The `state_info` had symbol keys like `:startedAt` and `:finishedAt` and we were looking for string keys like `"startedAt"` and `"finishedAt"` causing these values to always be `nil` even if data existed in the API response.

### CODE REVIEW

**Review by kbrock — APPROVED** [review]

**Comment by kbrock** [comment]
@Fryguy Is it possible to raise the priority of this bugfix? Or are we thinking the deleted_on logic is not critical?

File: spec/models/manageiq/providers/kubernetes/container_manager/refresher_spec.rb [hunk 1]
```diff
@@ -114,7 +114,7 @@ def assert_specific_container
           @container.last_started_at,
           @container.last_finished_at,
         ].each do |date_|
-          expect(date_.kind_of?(ActiveSupport::TimeWithZone) || date_.kind_of?(NilClass)).to be_truthy
+          expect(date_.kind_of?(ActiveSupport::TimeWithZone))
```
Comment (Fryguy) [inline spec/models/manageiq/providers/kubernetes/container_manager/refresher_spec.rb:117]: Was this change intentional?  This expectation is no longer expecting anything?

Comment (Fryguy) [inline spec/models/manageiq/providers/kubernetes/container_manager/refresher_spec.rb:117]: Was this change intentional?  This expectation is no longer expecting anything?

```suggestion
          expect(date_.kind_of?(ActiveSupport::TimeWithZone)).to be
```

or

```suggestion
          expect(date_.kind_of?(ActiveSupport::TimeWithZone)).to be_truthy
```

File: app/models/manageiq/providers/kubernetes/inventory/parser/container_manager.rb [hunk 2]
```diff
@@ -853,10 +853,10 @@ def parse_container_state(state_hash)
     return {} if state_hash.to_h.empty?
     res = {}
     # state_hash key is the state and value are attributes e.g 'running': {...}
-    (state, state_info), = state_hash.to_h.to_a
+    (state, state_info), = state_hash.to_h.deep_symbolize_keys.to_a
     res[:state] = state
     %w(reason started_at finished_at exit_code signal message).each do |attr|
```
Comment (Fryguy) [inline app/models/manageiq/providers/kubernetes/inventory/parser/container_manager.rb:858]: Maybe not for this PR, but we can pre-symbolize everything and not need to the .to_sym calls - Something like:

```suggestion
    %i(reason reason started_at startedAt finished_at finishedAt exit_code exitCode signal signal message message).each_slice(2) do |attr, state_info_attr|
```

**Comment by Fryguy** [comment]
@kbrock we'll backport to petrosian as part of the set, yes.  Even so, we'll need to deal with existing records anyway.

File: spec/models/manageiq/providers/kubernetes/container_manager/refresher_spec.rb (hunk 1 above)
Comment (agrare) [reply spec/models/manageiq/providers/kubernetes/container_manager/refresher_spec.rb:117]: I'm actually just going to drop the whole loop since `expect time or nil` is what hid this bug in the first place

File: app/models/manageiq/providers/kubernetes/inventory/parser/container_manager.rb (hunk 2 above)
Comment (kbrock) [reply app/models/manageiq/providers/kubernetes/inventory/parser/container_manager.rb:858]: yea, instead of each_slice, just using a hash is easier for me to read. But both work

Comment (agrare) [reply app/models/manageiq/providers/kubernetes/inventory/parser/container_manager.rb]: Done

**Comment by kbrock** [comment]
Rebase please. I think kubernetes HEAD fixes this issue

**Comment by Fryguy** [comment]
Backported to `petrosian` in commit 7793ce09194180ee171d172d73f9de22c95afdbb.

```text
commit 7793ce09194180ee171d172d73f9de22c95afdbb
Author: Keenan Brock <keenan@thebrocks.net>
Date:   Fri Jul 14 16:09:47 2023 -0400

    Merge pull request #496 from agrare/fix_started_at_finished_at_timestamps
    
    Fix started_at/finished_at always being nil
    
    (cherry picked from commit 1c753fe979d1ecec638b211d88f9ff7af8d8ed92)
```
